particle_structs/test/test_structure.cpp

```diff
@@ -357,6 +351,29 @@ int testSegmentComp(const char* name, PS* structure) {
   return fails;
 }
 
+int testPIDs(const char* name, PS* structure) {
+  printf("testPIDs %s, rank %d\n", name, comm_rank);
+  kkLidView pids;
+  kkLidView offsets;
+  int fails = 0;
+  structure->getPIDs(pids, offsets);
+
+  kkLidView failures("failures", 1);
+  kkLidView unsortedElems("unsortedElems", structure->capacity());
+  auto setUnsortedElems = PS_LAMBDA(const lid_t& e, const lid_t& p, const bool& mask) {
+    if (mask) unsortedElems(p) = e;
+  };
+  pumipic::parallel_for(structure, setUnsortedElems, "setUnsortedElems");
+
+  Kokkos::parallel_for(pids.size(), KOKKOS_LAMBDA(const lid_t& i) {
+    lid_t pid = pids(i);
+    lid_t oldElem = unsortedElems(pid);
+    assert(i >= offsets(oldElem));
+  });
+  fails += pumipic::getLastValue(failures);
+  return fails;
+}
+
```

Comment:
@cwsmith I am not sure if this is an adequate test of the functionality. What do you think?

Reply:
Looks good.  We could tighten up the assert to:

```
assert(i >= offsets(oldElem) && i < offsets(oldElem+1));
```

Should the assert be replaced with a conditional that sets the `failures` array?

Reply:
Yeah good catch